instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/server/RemoteServerInstrumentation.java

```diff
@@ -43,38 +42,45 @@ public void transform(TypeTransformer transformer) {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
     public static void onEnter(
-        @Advice.Origin Method method,
+        @Advice.Origin("#t") Class<?> declaringClass,
+        @Advice.Origin("#m") String methodName,
         @Advice.Local("otelCallDepth") CallDepth callDepth,
+        @Advice.Local("otelRequest") ClassAndMethod request,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
+
       callDepth = CallDepth.forClass(RemoteServer.class);
       if (callDepth.getAndIncrement() > 0) {
         return;
       }
 
       // TODO review and unify with all other SERVER instrumentation
       Context parentContext = THREAD_LOCAL_CONTEXT.getAndResetContext();
+      request = ClassAndMethod.create(declaringClass, methodName);
```

Comment:
You can get an instance of `Method` here, then you don't need a separate `ClassAndMethod` class

Reply:
ByteBuddy returns a new `Method` instance each time the instrumentation is executed -- method name and class are constants. I believe @laurit has found that issue, I can't find the PR for that though